src/OrleansProviders/Streams/Memory/MemoryMessageBody.cs

```diff
@@ -29,26 +30,35 @@ public interface IMemoryMessageBodySerializer
     /// Default IMemoryMessageBodySerializer
     /// </summary>
     [Serializable]
-    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer
+    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer, IOnDeserialized
     {
+        [NonSerialized]
+        private SerializationManager serializationManager;
+
         /// <summary>
-        /// Serialize MemoryMessageBody to an array segment of bytes.
+        /// Initializes a new instance of the <see cref="DefaultMemoryMessageBodySerializer"/> class.
         /// </summary>
-        /// <param name="body"></param>
-        /// <returns></returns>
+        /// <param name="serializationManager"></param>
+        public DefaultMemoryMessageBodySerializer(SerializationManager serializationManager)
+        {
+            this.serializationManager = serializationManager;
+        }
+
+        /// <inheritdoc />
         public ArraySegment<byte> Serialize(MemoryMessageBody body)
         {
-            return new ArraySegment<byte>(SerializationManager.SerializeToByteArray(body));
+            return new ArraySegment<byte>(serializationManager.SerializeToByteArray(body));
         }
 
-        /// <summary>
-        /// Deserialize an array segment into a MemoryMessageBody
-        /// </summary>
-        /// <param name="bodyBytes"></param>
-        /// <returns></returns>
+        /// <inheritdoc />
         public MemoryMessageBody Deserialize(ArraySegment<byte> bodyBytes)
         {
-            return SerializationManager.DeserializeFromByteArray<MemoryMessageBody>(bodyBytes.ToArray());
+            return serializationManager.DeserializeFromByteArray<MemoryMessageBody>(bodyBytes.ToArray());
+        }
+
+        void IOnDeserialized.OnDeserialized(ISerializerContext context)
+        {
+            this.serializationManager = context.SerializationManager;
```

Comment:
correct me if I'm wrong, but with the latest, this class is never (de)serialized, so the IOnDeserialized interface is not needed here. It will always be constructed using ActivatorUtilities. In theory you could also drop the [Serializable] attribute from it (it wasn't being used before either). Unless you added this in case someone was *incorrectly* using it and serializing it for a different purpose?